### DISCUSSION TITLE
How to get IsNew/Parent in BusinessRule w/o using Target?

### DISCUSSION THREAD

**Discussion by ossendorf-at-hoelscher** [MarimerLLC/csla#3634] [Questions]
Hello.

We have some rules which should only be executed when the BO is either New or not. And in some rare cases we have to inspect the childs parent type to decide further checks.

Is this possible without using the `Target` property of the `IRuleContext`? As far as I'm aware this is somewhat non problematic in sync-rules.  But since async rules are more and more needed with the `Target` not provided by default. We want to - if possible - avoid using it at all.

Thanks in advance

**Comment by rockfordlhotka** [comment]
Right now the rules engine has no way for you to access the `Parent` property without the `Target` property on the `context` parameter.

You are right, in a sync rule this is straightforward. In an async rule it is challenging because there are concurrency issues to think through.

It would be an excellent enhancement to CSLA to have an API that allows safe access to properties from the entire object graph in an async rule. I have never had the time to think through the design for that API, but it would be fun!

**Reply by ossendorf-at-hoelscher** [reply]
Thanks for your reply.

Parent I somewhat expected that. How about the `IsNew` flag? Is there a way to get that in the rule without the `Target`?

Maybe it's out of scope for this question: But why is concurrency an issue in an async rule? Are the rules actually executed in parallel? Otherwise I don't see a Problem there.
